src/BatchDealImportService.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Drupal\amazon_product_widget;
+
+use Drupal\amazon_product_widget\Exception\DealFeedFinishedProcessingException;
+
+/**
+ * Class BatchDealImportService.
+ *
+ * @package Drupal\amazon_product_widget
+ */
+class BatchDealImportService {
+
+  /**
+   * Imports deals into storage from CSV.
+   *
+   * @param string $filename
+   *   Filename from where to import.
+   * @param int $total
+   *   The total entries in the CSV file.
+   * @param $context
+   *   The context.
+   */
+  public static function importChunked(string $filename, int $total, &$context) {
+    /** @var \Drupal\amazon_product_widget\DealFeedService $dealFeedService */
+    $dealFeedService = \Drupal::service('amazon_product_widget.deal_feed_service');
+    $maxProcessingTime = $dealFeedService->getMaxProcessingTime();
+
+    // How many imports to do per call to DealFeedService::import().
+    $importsPerRound = 1000;
+
+    if (empty($context['sandbox'])) {
+      $context['sandbox']['filename'] = $filename;
+      $context['sandbox']['total'] = $total;
+      $context['sandbox']['processed'] = 0;
+    }
+
+    $timeStart = time();
+    while (TRUE) {
+      try {
+        $context['sandbox']['errors'] += $dealFeedService->import(
+          $context['sandbox']['filename'],
+          $context['sandbox']['processed'],
+          $importsPerRound
+        );
+        $context['sandbox']['processed'] += $importsPerRound;
+        $context['results']['processed'] = $context['sandbox']['processed'];
+        $context['results']['errors'] = $context['sandbox']['errors'];
+      }
+      catch (DealFeedFinishedProcessingException $exception) {
+        $context['finished'] = 1;
+        break;
+      }
+
+      if ($context['sandbox']['errors'] >= $dealFeedService->getMaxDealImportErrors()) {
+        $context['finished'] = 1;
+        break;
+      }
+
+      $timeCurrent = time();
+      if ($timeCurrent - $timeStart >= $maxProcessingTime) {
+        $batch = &batch_get();
+        $batch_next_set = $batch['current_set'] + 1;
+        $batch_set = &$batch['sets'][$batch_next_set];
+        $batch_set['operations'][] = [
+          '\Drupal\amazon_product_widget\BatchDealImportService::importChunked',
+        ];
+
+        $batch_set['total'] = count($batch_set['operations']);
```

Comment:
i dont understand this

Reply:
If `$batch_set['total']` and `$batch_set['count']`are not set, the batch immediately finishes and exits. Since we are just adding one operation always, it it practically always 1. This could be simplified to `$batch_set['total'] = $batch_set['count'] = 1;`